File: Front/nexta/src/shared/entities/Order.ts
import { BasketProduct } from "../../models/basketProduct/BasketProduct";
import { Product } from "../../models/Product";
import { User } from "./User";

export interface Order{
	id:string,
    userId:string,
	user:User,
	basketProducts: BasketProduct[]
	products:Product[]
	status:OrderStatus,
	createdDate: string
}

export enum OrderStatus{
	Unknown = -1, //Неизвестный статус
	Accepted = 0, //Принят
	InProgress = 1, //В работе
	Canceled = 2, //Отменён
	Ready = 3, //Готов к выдаче
	Complete = 4 //Завершён
}

File: Back/Nexta/Nexta.Domain/Models/Order.cs
using Nexta.Domain.Enums;

namespace Nexta.Domain.Models
{
    public class Order
    {
		public Guid Id { get; set; }
		public Guid UserId { get; set; }
		public User User { get; set; } = null!;

		public List<Product>? Products { get; set; }
		public List<OrderProduct>? OrderProducts { get; set; }

		public DateOnly CreatedDate { get; set; }
		public OrderStatus Status { get; set; }
	}
}